sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffleCoalesceExec.scala

```diff
@@ -233,24 +236,33 @@ class KudoTableOperator(kudo: Option[KudoSerializer])
   extends SerializedTableOperator[KudoSerializedTableColumn] {
   require(kudo != null, "kudo serializer should not be null")
 
-  override def getDataLen(column: KudoSerializedTableColumn): Long = column.kudoTable.getHeader
+  override def getDataLen(column: KudoSerializedTableColumn): Long =
+    column.spillableKudoTable.getHeader
     .getTotalDataLen
 
-  override def getNumRows(column: KudoSerializedTableColumn): Int = column.kudoTable.getHeader
+  override def getNumRows(column: KudoSerializedTableColumn): Int =
+    column.spillableKudoTable.getHeader
     .getNumRows
 
   override def concatOnHost(columns: Array[KudoSerializedTableColumn]): CoalescedHostResult = {
     require(columns.nonEmpty, "no tables to be concatenated")
-    val numCols = columns.head.kudoTable.getHeader.getNumColumns
+    val numCols = columns.head.spillableKudoTable.getHeader.getNumColumns
     if (numCols == 0) {
       val totalRowsNum = columns.map(getNumRows).sum
       RowCountOnlyMergeResult(totalRowsNum)
     } else {
-
-      val kudoTables = columns.map(_.kudoTable)
-      val result = kudo.get.mergeOnHost(kudoTables)
-
-      KudoHostMergeResultWrapper(result)
+      try {
+        for (skt <- columns.map(_.spillableKudoTable)) {
```

Comment:
could a `SpillableKudoTable` make a `KudoTable` that isn't spillable? I don't know if `SpillableKudoTable` has to subclass `KudoTable`, but I feel this would clean this interface a bit.

So you say `skts.safeMap(_.makeKudoTable)`. `KudoTable` is already closeable, so when we close the sequence of `KudoTable` all `HostMemoryBuffer` references are closed and we are now spillable again. 

We then later safe `skts.safeClose()` to close the spillable handles.